### PR TITLE
11.0 mig account invoice constraint chronology

### PR DESCRIPTION
**Pull request by OpenCode**
Migrate account_invoice_constraint_chronology to Odoo 11.0

### CODE REVIEW

File: account_invoice_constraint_chronology/tests/test_account_constraint_chronology.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
```
Comment (etobella) [inline account_invoice_constraint_chronology/tests/test_account_constraint_chronology.py]: No longer necessary on py files

File: account_invoice_constraint_chronology/tests/test_account_constraint_chronology.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+# Copyright 2015-2017 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
```
Comment (etobella) [inline account_invoice_constraint_chronology/tests/test_account_constraint_chronology.py]: https

**Review by etobella — APPROVED** [review]
LGTM :+1: 
just two comments

**Comment by OpenCode** [comment]
@etobella Done! Thank you for your review :)

File: account_invoice_constraint_chronology/README.rst [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+=====================================
+Account Invoice Constraint Chronology
+=====================================
+
+This module helps ensuring the chronology of invoice numbers.
+
+It prevents the validation of invoices when:
+
+* there are draft invoices with an anterior date
```
Comment (JordiBForgeFlow) [inline account_invoice_constraint_chronology/README.rst:13]: replace 'anterior' with 'prior'

File: account_invoice_constraint_chronology/README.rst [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+* there are validated invoices with a posterior date
```
Comment (JordiBForgeFlow) [inline account_invoice_constraint_chronology/README.rst:14]: replace 'posterior' with 'later'

**Review by JordiBForgeFlow — CHANGES_REQUESTED** [review]

File: account_invoice_constraint_chronology/model/account_invoice.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+                                           limit=1)
+
+                    if invoices:
+                        date_invoice_format = fields.Date.\
+                            from_string(inv.date_invoice)
+                        date_invoice_tz = fields\
+                            .Date.context_today(self, date_invoice_format)
+                        raise UserError(_("Chronology Error. "
+                                          "There exist at least one invoice "
+                                          "with a date posterior to %s.") %
```
Comment (rgarnau) [inline account_invoice_constraint_chronology/model/account_invoice.py:49]: as @jbeficent mentions before, change 'posterior'.

File: account_invoice_constraint_chronology/README.rst [hunk 6]
```diff
@@ -0,0 +1,59 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
```
Comment (rgarnau) [inline account_invoice_constraint_chronology/README.rst:2]: https

File: account_invoice_constraint_chronology/README.rst [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+It prevents the validation of invoices when:
+
+* there are draft invoices with an anterior date
+* there are validated invoices with a posterior date
+
+Configuration
+=============
+
+To configure this module, go to the menu *Accounting > Configuration > Journals > Journals* and activate the option *Check Chronology* on the relevant journals. After the installation of the module, this option will be active on *sale* and *sale refund* journals.
```
Comment (rgarnau) [inline account_invoice_constraint_chronology/README.rst:19]: Replace *Accounting > Configuration > Journals > Journals*
for *Accounting/Invoicing > Configuration > Accounting > Journals*

**Review by rgarnau — CHANGES_REQUESTED** [review]

**Comment by JordiBForgeFlow** [comment]
@OpenCode Do you prefer us to send a PR with these minor fixes suggested?

**Comment by JordiBForgeFlow** [comment]
As @OpenCode is not actively managing the PR we have built on top of his branch to correct the minor fixed and proposed the PR https://github.com/OCA/account-financial-tools/pull/600, so closing this one.

**Comment by OpenCode** [comment]
@jbeficent I was working on requestes. I saw @rgarnau solved it with another PR. Ok.

**Pull request closed without merging**